apps/wing/src/analytics/storage.ts

```diff
@@ -0,0 +1,202 @@
+    this.analyticsStorageDir = props?.analyticsStorageDir ?? path.join(os.tmpdir(), "wing-analytics");
+    this.debug = props?.debug;
+    this.analyticsConfig = this.loadConfig();
+  }
+
+  /**
+   * Stores a single analytic event to disk
+   * 
+   * @param event the analytic event to save
+   * @returns the path to the saved event or undefined if there was an error
+   */
+  public storeAnalyticEvent(event: AnalyticEvent): string | undefined {
+    try {
+      if (this.analyticsConfig.optOut == true) {
+        return undefined;
+      }
+      const eventId = uuidv4();
+      
+      if (!existsSync(this.analyticsStorageDir)) {
+        mkdirSync(this.analyticsStorageDir);
+      }
+  
+      const analyticReportFile = path.join(this.analyticsStorageDir, `${eventId}.json`);
+      
+      
+      // attach timestamp and anonymousId to event
+      event.timestamp = event.timestamp ?? new Date().toISOString();
+      
+      // We add the anonymousId only if we are not in a CI environment
+      if (event.properties.ci) {
+        const anonymousId = this.getAnonymousId();
+        event.anonymousId = anonymousId;
+      }
+
+      this.saveEvent(analyticReportFile, event);
+    
+      return analyticReportFile;
+  
+    } catch (error) {
+      return undefined;
+    }
+  }
+
+  /**
+   * Retrieves an analytic event from disk
+   * 
+   * @param filePath the path the event was saved to
+   * @returns the event or undefined if there was an error
+   */
+  public loadEvent(filePath: string): AnalyticEvent | undefined {
+    try {
+      const fileContents = readFileSync(filePath, 'utf-8');
+      return JSON.parse(fileContents) as AnalyticEvent;
+    } catch (error) {
+      // ignore
+      return undefined;
+    }
+  }
+
+  /**
+   * Reads the analytics config for the user's anonymous id, 
+   * if an id does not exist, one is generated and saved to disk
+   * 
+   * @returns the anonymous id for the user
+   */
+  public getAnonymousId(): string {
+    let config = this.loadConfig();
+    if (!config.anonymousId) {
+      config.anonymousId = this.generateAnonymousId();
+      this.saveConfig(config);
+    }
+    return config.anonymousId;
+  }
+
+
+  private generateAnonymousId(): string {
+    return randomBytes(16).toString('hex');
+  }
+
+  /**
+   * Retrieves the analytics config from disk, if one does not exist
+   * a new one is created and saved to disk
+   * 
+   * @returns the analytics config for the user
+   */
+  public loadConfig(): AnalyticsConfig {
+    try {
+      const fileContents = readFileSync(this.analyticsConfigFile, 'utf-8');
+      return JSON.parse(fileContents) as AnalyticsConfig;
+    } catch (error: any) {
+      // Possible issue reading analytics config file, first determine if file exists yet
+  
+      if (error.code === 'EACCES') {
+        throw new Error("Not able to access analytics config file due to permissions");
```

Comment:
Is this caught somewhere? I don't think we need this to ever fail even if we can't store the file